blueocean-dashboard/src/main/js/components/CellLink.js

```diff
@@ -0,0 +1,42 @@
+import React, { PropTypes } from 'react';
+import { Link } from 'react-router';
+
+/**
+ * A convenience wrapper to pass down 'linkUrl' to child CellLink components.
+ */
+export class CellRow extends React.Component {
+
+    render() {
+        return (
+            <tr>
+                {React.Children.map(
+                    this.props.children,
+                    child => React.cloneElement(child, { linkUrl: this.props.linkUrl })
+                )}
+            </tr>
+        );
+    }
+}
+
+CellRow.propTypes = {
+    children: PropTypes.node,
+    linkUrl: PropTypes.string,
+};
+
+/**
+ * A table cell with an embedded react-router link.
+ */
+export function CellLink(props) {
+    const url = props.linkUrl || '';
+
+    return (
+        <td className="tableRowLink">
+            <Link to={url}>{props.children}</Link>
```

Comment:
well we have a couple of resolver in place ATM browser and query (that can be extended to cookies, localStore, ...). However since we do not use react-i18next anymore we would need to implement a reload on language changes (e.g. browser preferred language depends on that).

Anyway it is a general problem in our code base. Any link that is not passing the current query is removing it from the url.

For example if we implement filters or search they would use query parameter to instruct underlying code to filter results. ```?filter=type&search=multibranch``` if we do not pass this query to the links we remove this filters/search context. 

Meaning we need to be very careful when we remove the query from the url.